File: app/src/main/java/org/movieos/feeder/MainActivity.java
package org.movieos.feeder;

import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.webkit.WebView;

import org.movieos.feeder.fragment.EntriesFragment;
import org.movieos.feeder.fragment.LoginFragment;
import org.movieos.feeder.utilities.FragmentBackHandler;
import org.movieos.feeder.utilities.Settings;
import org.movieos.feeder.utilities.SyncTask;

import io.realm.Realm;

public class MainActivity extends AppCompatActivity {

    Snackbar mSnackbar;
    // control the lifetime of the realm object
    Realm mRealm;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if (savedInstanceState == null) {
            getSupportFragmentManager()
                .beginTransaction()
                .add(R.id.main_content, Settings.getCredentials(this) != null ? new EntriesFragment() : new LoginFragment())
                .commitNow();
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        mRealm = Realm.getDefaultInstance();
        new WebView(this).loadData("test", null, null);
        SyncTask.sync(this, false, false);
    }

    @Override
    protected void onStop() {
        super.onStop();
        mRealm.close();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    @Override
    public void onBackPressed() {
        Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.main_content);
        if (fragment instanceof FragmentBackHandler) {
            boolean handled = ((FragmentBackHandler) fragment).onBackPressed();
            if (handled) {
                return;
            }
        }
        super.onBackPressed();
    }

}

File: app/src/main/java/org/movieos/feeder/MainActivity.kt
package org.movieos.feeder

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.webkit.WebView
import com.google.android.gms.auth.api.Auth
import com.google.android.gms.common.ConnectionResult
import com.google.android.gms.common.api.GoogleApiClient
import io.realm.Realm
import org.movieos.feeder.fragment.EntriesFragment
import org.movieos.feeder.fragment.LoginFragment
import org.movieos.feeder.utilities.Settings
import org.movieos.feeder.utilities.SyncTask

class MainActivity : AppCompatActivity(), GoogleApiClient.OnConnectionFailedListener {

    // control the lifetime of the realm object
    internal var realm: Realm? = null

    val client by lazy {
        GoogleApiClient.Builder(this)
                .enableAutoManage(this, this)
                .addApi(Auth.CREDENTIALS_API)
                .build()
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            supportFragmentManager
                    .beginTransaction()
                    .add(R.id.main_content, if (Settings.getCredentials(this) != null) EntriesFragment() else LoginFragment())
                    .commit()
        }
    }

    override fun onStart() {
        super.onStart()
        realm = Realm.getDefaultInstance()
        WebView(this).loadData("test", null, null)
        SyncTask.sync(this, false, false)
    }

    override fun onStop() {
        super.onStop()
        SyncTask.sync(this, false, true)
        realm?.close()
        realm = null
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    override fun onConnectionFailed(p0: ConnectionResult) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        val fragment = supportFragmentManager.findFragmentById(R.id.main_content)
        fragment.onActivityResult(requestCode, resultCode, data)
    }
}
